### DISCUSSION TITLE
Request for an option in the Search Results to not execute a search without a value

### DISCUSSION THREAD

**Discussion by GilesGregg** [microsoft-search/pnp-modern-search#2840] [Ideas]
**Is your feature request related to a problem? Please describe.**
When connected to the Search Box Web part, the search results web part executes a search by default with no query text, returns unpredictable results

**Describe the solution you'd like**
An option within the Search Results web part - do not execute a search unless a non-blank value is provided by the Search Box web part

**Describe alternatives you've considered**
Using the default text option, using a phrase which returns no results

**Additional context**
.

**Comment by wobba** [comment]
I suspect you have additions in your query template? The workaround is to add that using FQL in the `Refinement filters` section instead.

**Comment by GilesGregg** [comment]
Could you be more specific in what to add in the refinement filters section? The ideal is for the Search Results web part not to execute unless it is provided with a search term.

**Comment by wobba** [comment]
Translate the KQL you have in your query template to FQL and put it into the refinement filter section. And leave the query template as `{searchTerms}`.

https://learn.microsoft.com/en-us/sharepoint/dev/general-development/fast-query-language-fql-syntax-reference

**Comment by GilesGregg** [comment]
And that will stop the search results web part from executing a query? Will try.

**Comment by GilesGregg** [comment]
I've moved the path parameter to the refinement panel, the web part is still executing with no query terms
![searcherror](https://user-images.githubusercontent.com/53594934/204425181-021cf2df-df88-4e8b-a59a-ecb4489079de.png)

**Comment by wobba** [comment]
I cannot repro. I have a sb webpart connect to the result web part with the exact same setup (and make sure you have no default query on the dynamic data setup). When the search box is empty I have 0 results. When I add a query I get results. Using v4.8 right now.

**Comment by GilesGregg** [comment]
Was this fixed in 4.8? I'm on 4.7 waiting for my firm to upgrade

**Comment by wobba** [comment]
@GilesGregg do you have a test tenant you can use to verify? And no reason to use FQL in the refinement filters as you can type the same as KQL in the query template. See if that makes a change for you.
